src/main/java/com/epam/vsharstuk/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.epam.vsharstuk.service.impl;
+
+import com.epam.vsharstuk.controller.CarPageController;
+import com.epam.vsharstuk.dao.CarRepository;
+import com.epam.vsharstuk.model.Car;
+import com.epam.vsharstuk.service.CarSearchCriteria;
+import com.epam.vsharstuk.service.CarService;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CarServiceImpl implements CarService {
+
+    @Autowired
+    private CarRepository carRepository;
+
+    private Logger LOG = Logger.getLogger(CarPageController.class);
+
+    @Override
+    public void addCar(String make, String model, String year, String cost, Integer userId) {
+        Car car = new Car();
+        car.setMake(make);
+        car.setModel(model);
+        car.setYear(Integer.valueOf(year));
+        car.setCost(Integer.valueOf(cost));
+        car.setUserId(userId);
+        carRepository.addCar(car);
+        LOG.info("Car was added: " + car);
+    }
+
+    @Override
+    public List<Car> findCarByMake(String make, String order) {
+
+        if (make == null) {
+            return Collections.emptyList();
+        }
+
+        List<Car> cars = carRepository.findCarByMake(make);
+
+        if (order != null ) {
+            return sort(cars, order);
+        }
+        return cars;
+    }
+
+    @Override
+    public List<Car> findCarByCriteria(CarSearchCriteria criteria, String order) {
+
+        if (criteria == null) {
+            return Collections.emptyList();
+        }
+
+        List<Car> cars = carRepository.findCarByCriteria(criteria);
+
+        if (order != null ) {
+            return sort(cars, order);
+        }
+        return cars;
+    }
+
+    private List<Car> sort(List<Car> cars, String order) {
```

Comment:
This can be implemented in one line like:  
```
private List<Car> sort(List<Car> cars, Function<Car, String> keyExtractor) {
     return return cars.stream().sorted(Comparator.comparing(keyExtractor)).collect(Collectors.toList());
}
```  
and used:  
```  
carService.sort(cars, car -> car.getModel());
```